bots/run-queue

```diff
@@ -26,11 +26,64 @@ import ssl
 import subprocess
 import sys
 import time
+import logging
 
 import pika
 from task import redhat_network
 
+# Returns a command to execute
+def consume_webhook_queue(channel, amqp):
+    # interpret payload
+    # call tests-scan or issue-scan appropriately
+    method_frame, header_frame, body = channel.basic_get(queue='webhook')
+    if not method_frame:
+        # No message available
+        logging.warn('No message in the webhook queue available')
+        return None
 
+    body = json.loads(body)
+    event = body['event']
+    request = body['request']
+    if event == 'pull_request':
+        pull_request = request['pull_request']
+        repo = pull_request['base']['repo']['full_name']
+        cmd = ['bots/tests-scan', '--pull-data', json.dumps(pull_request), '--amqp', opts.amqp]
+    elif event == 'status':
+        repo = request['repository']['full_name']
+        sha = request['sha']
+        context = pipes.quote(request['context'])
+        cmd = ['bots/tests-scan', '--sha', sha, '--amqp', opts.amqp, '--context', context]
+    else:
+        logging.error('Unkown event type in the webhook queue')
+        return None
+
+    # the cockpit repo itself is treated as a special case in tests-scan, in
```

Comment:
We usually prefix such things with "HACK:", so that they are greppable.

Reply:
ack